pinecone/index_connection.go

```diff
@@ -46,7 +47,7 @@ func (idx *IndexConnection) Close() error {
 	return err
 }
 
-func (idx *IndexConnection) UpsertVectors(ctx *context.Context, in []*Vector) (uint32, error) {
+func (idx *IndexConnection) UpsertVectors(ctx context.Context, in []*Vector) (uint32, error) {
```

Comment:
I think this is fine, especially given `akCtx` returns by value. At least that's my understanding.

Reply:
Double check and make sure that it isn't doing a copy on the stack. The idea is you should be able to pass a Context to different function calls and do a "Cancel" on the single context to abort all the in-process calls.

If it's copying the Context instead of passing a reference that wouldn't work.